library/core/src/iter/traits/iterator.rs

```diff
@@ -3434,11 +3536,18 @@ pub trait Iterator {
     #[stable(feature = "iter_order", since = "1.5.0")]
     fn cmp<I>(self, other: I) -> Ordering
     where
-        I: IntoIterator<Item = Self::Item>,
-        Self::Item: Ord,
+        I: ~const IntoIterator<Item = Self::Item>,
+        Self::Item: ~const Ord + ~const Destruct,
+        I::IntoIter: ~const Destruct,
+        I::Item: ~const Destruct,
+        Self::Item: ~const Destruct,
+        Self: ~const Destruct,
```

Comment:
Will these show up in rustdoc?

Reply:
I don't think so but I will check later

Reply:
Haven't got the time to come back to this yet, but [this](https://doc.rust-lang.org/std/primitive.bool.html#impl-bool) is pretty good evidence that rustdoc hides these bounds. All non `~const Destruct` bounds are displayed without `~const`, and `~const Destruct` bounds are all hidden.